sources/input/scene_open.c

```diff
@@ -0,0 +1,39 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   scene_open.c                                       :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: lfarias- <lfarias-@student.42.rio>         +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2023/02/13 02:16:42 by lfarias-          #+#    #+#             */
+/*   Updated: 2023/03/11 19:09:29 by lfarias-         ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "../../headers/mini_rt.h"
+#include <errno.h>
+#include <unistd.h>
+#include <fcntl.h>
+
+int	scene_open(char *filename)
+{
+	int	scene_fd;
+
+	scene_fd = open(filename, O_WRONLY);
+	if (scene_fd < 0 && errno == EISDIR)
+	{
+		print_wrong_extension();
```

Comment:
Why print_wrong_extension?

Reply:
print_wrong_extension prints the following message:

ERROR: the current filename is not a .rt file

because a directory can't ever be a proper .rt file I think this message fits really well.

but the function name is really ambiguous given this specific type of error. Maybe a refactor to `print_invalid_file` is a solution. what do you think?